extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/MTLS.java

```diff
@@ -0,0 +1,196 @@
+package io.quarkus.vertx.http.security;
+
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.AuthRuntimeConfig;
+import io.quarkus.vertx.http.runtime.VertxHttpBuildTimeConfig;
+import io.quarkus.vertx.http.runtime.VertxHttpConfig;
+import io.quarkus.vertx.http.runtime.security.CertificateRoleAttribute;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.quarkus.vertx.http.runtime.security.MtlsAuthenticationMechanism;
+import io.smallrye.common.annotation.Experimental;
+import io.smallrye.config.SmallRyeConfigBuilder;
+import io.vertx.core.http.ClientAuth;
+
+/**
+ * This class provides a way to create the mutual TLS client authentication mechanism. The {@link HttpAuthenticationMechanism}
+ * created with this class can be registered using the {@link HttpSecurity#mechanism(HttpAuthenticationMechanism)} method.
+ */
+@Experimental("This API is currently experimental and might get changed")
+public interface MTLS {
+
+    /**
+     * Creates the mutual TLS client authentication mechanism.
+     * The client authentication is accepted if presented by a client.
+     * Use this method if the client authentication is only required for certain routes
+     * and secure these routes with HTTP permissions or standard security annotations.
+     *
+     * @return HttpAuthenticationMechanism
+     * @see VertxHttpBuildTimeConfig#tlsClientAuth() for more information
+     */
+    static HttpAuthenticationMechanism request() {
```

Comment:
Should there be overloads for users to choose the matching TLS configuration name ?

Reply:
Or may not overloads, but a dedicated method like `.tls(String name)` ?

Reply:
I had a quick look and I think adding the `tls(String name)` should be doable without big changes and I thought about how to make it even safer. I'll try it when the other PR of mine gets merge to avoid merge conflicts.